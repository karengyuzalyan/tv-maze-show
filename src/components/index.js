export { InfoWrapper} from './info-wrapper';
export { BreadCrumbs } from './breadcrumbs';
export { Loader } from './loader';