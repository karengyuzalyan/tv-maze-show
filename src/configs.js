export const configs = {
    API_URL: 'http://api.tvmaze.com/',
    SHOW_ID: 6771,
    BASE_URL: 'http://localhost:3000/'
}