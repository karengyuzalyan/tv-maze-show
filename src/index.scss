$show-title-color: #2cb5ba;

// The m-y-1 means
// Margin by y-coordinates(Top and Bottom) and the number 2 means 1 rem.
.m-y-2 {
  margin: 2rem 0;
}
.m-y-1 {
  margin: 1rem 0;
}
.m-b-1 {
  margin: 0 0 1rem 0;
}
.m-b-2 {
  margin: 0 0 2rem 0;
}

a {
  color: #6bd2cc !important;
  &:hover{
      text-decoration: none !important;
  }
}

.show-title-color {
  color: $show-title-color;
}

.hidden-sm-up {
  display: none;
}

.t-center {
  text-align: center;
}

.d-flex {
  display: flex
}

.f-column {
  flex-direction: column;
}

.j-between {
  justify-content: space-between;
}

.a-center {
  align-items: center;
}

.size-24 {
  font-size: 24px;
}

$breakpoint-tablet: 768px;
@media (max-width: $breakpoint-tablet) {
  .hidden-sm-down {
    display: none;
  }

  .hidden-sm-up {
    display: block;
  }
}