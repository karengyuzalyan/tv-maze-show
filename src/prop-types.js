import PropTypes from 'prop-types';

const ImagePropTypes = PropTypes.shape({
    original: PropTypes.string,
    medium: PropTypes.string,
});

const HeadersPropTypes = PropTypes.shape({
    'cache-control': PropTypes.string,
    'content-type': PropTypes.string,
});

export const EpisodePropTypes = PropTypes.shape({
    airstamp: PropTypes.string,
    summary: PropTypes.string,
    airdate: PropTypes.string,
    runtime: PropTypes.number,
    airtime: PropTypes.string,
    number: PropTypes.number,
    season: PropTypes.number,
    name: PropTypes.string,
    url: PropTypes.string,
    image: ImagePropTypes,
    id: PropTypes.number,
});

export const EsisodesListPropTypes = PropTypes.shape({
    data: PropTypes.arrayOf(EpisodePropTypes),
    statusText: PropTypes.string,
    headers: HeadersPropTypes,
    config: PropTypes.shape(),
    status: PropTypes.number,
});

export const ShowDetailsPropTypes = PropTypes.shape({
    data: PropTypes.shape({
        genres: PropTypes.arrayOf(PropTypes.string),
        webChannel: PropTypes.shape({}),
        officialSite: PropTypes.string,
        externals: PropTypes.shape({}),
        schedule: PropTypes.shape({}),
        network: PropTypes.shape({}),
        average: PropTypes.shape({}),
        premiered: PropTypes.string,
        rating: PropTypes.shape({}),
        language: PropTypes.string,
        runtime: PropTypes.number,
        weight:  PropTypes.number,
        summary: PropTypes.string,
        updated: PropTypes.number,
        status: PropTypes.string,
        type: PropTypes.string,
        name: PropTypes.string,
        url: PropTypes.string,
        image: ImagePropTypes,
        id: PropTypes.number,
    }),
    config: PropTypes.shape({}),
    statusText: PropTypes.string,
    status: PropTypes.number,
    headers: HeadersPropTypes,
});
