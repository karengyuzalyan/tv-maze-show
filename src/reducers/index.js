import { combineReducers } from 'redux'
import showData from './show-details';
import episodesList from './episodes-list';
import episode from './episode';

export default combineReducers({
  episode,
  showData,
  episodesList,
})