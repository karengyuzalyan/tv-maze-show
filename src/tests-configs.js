import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

// Configuration of enzyme
Enzyme.configure({ adapter: new Adapter() });

export * from 'enzyme';