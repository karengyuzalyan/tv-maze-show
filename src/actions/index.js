import base from 'axios';
import { configs } from 'src/configs';
import {
    GET_SHOW_BY_ID_SUCCESS,
    GET_SHOW_BY_ID_FAIL,
    GET_EPISODE_BY_NUMBER_SUCCESS,
    GET_EPISODE_BY_NUMBER_FAIL,
    GET_EPISODES_BY_SHOW_SUCCESS,
    GET_EPISODES_BY_SHOW_FAIL,
  } from './types';

const axios = base.create({
  baseURL: configs.apiUrl
});

// The Function bellow received the ID argument for getting the Show information.
export function getShowByID(ID) {
    return dispatch => {
        axios.get(`/shows/${ID}`)
        .then(res => {
            dispatch({ type: GET_SHOW_BY_ID_SUCCESS, data: res });
        })
        .catch(err => {
            dispatch({ type: GET_SHOW_BY_ID_FAIL, payload: err });
        });
    };
}

// The Function bellow received the query for getting the Episode information by Show, Season and Number of episode.
export function getEpisodeByNumber(query) {
    return dispatch => {
        axios.get(`${query}`)
        .then(res => {
            dispatch({ type: GET_EPISODE_BY_NUMBER_SUCCESS, data: res });
        })
        .catch(err => {
            dispatch({ type: GET_EPISODE_BY_NUMBER_FAIL, payload: err });
        });
    };
}

// The Function bellow received the ID argument for getting the Episodes list by Show ID.
export function getEpisodesByShow(ID) {
    return dispatch => {
        axios.get(`/shows/${ID}/episodes`)
        .then(res => {
            dispatch({ type: GET_EPISODES_BY_SHOW_SUCCESS, data: res });
        })
        .catch(err => {
            dispatch({ type: GET_EPISODES_BY_SHOW_FAIL, payload: err });
        });
    };
}