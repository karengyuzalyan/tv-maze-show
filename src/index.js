import React from 'react';
import { createStore, applyMiddleware } from 'redux';
import reduxThunk from 'redux-thunk';
import { Provider } from 'react-redux';
import ReactDOM from 'react-dom';
import Routes from './routes';
import reducers from './reducers';
import * as serviceWorker from './serviceWorker';
import './index.scss';

const middlewares = [reduxThunk];
const createStoreWithMiddleware = applyMiddleware(...middlewares)(createStore);
const store = createStoreWithMiddleware(reducers);

ReactDOM.render(
    <Provider store={store}>
        <Routes />
    </Provider>, document.getElementById('root'));

// If you want your app to work offline and load faster, you can change
// unregister() to register() below. Note this comes with some pitfalls.
// Learn more about service workers: http://bit.ly/CRA-PWA
serviceWorker.unregister();
