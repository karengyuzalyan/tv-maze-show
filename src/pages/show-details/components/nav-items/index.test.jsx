// Module imports
import React from 'react';
import { shallow } from 'src/tests-configs';

// Internal imports
import { NavItems } from './index';

const props = {
    seasons: ["1", "2"],
    activeTab: 1,
};

describe('src/nav-items/index.jsx', () => {
  describe('<NavItems />', () => {
    test('should render and match snapshot', () => {
      const component = shallow(<NavItems {...props} />)
      expect(component).toMatchSnapshot()
    });
  });
});
