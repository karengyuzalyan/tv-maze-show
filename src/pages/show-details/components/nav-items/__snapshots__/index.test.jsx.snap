// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/nav-items/index.jsx <NavItems /> should render and match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NavItems
    activeTab={1}
    seasons={
      Array [
        "1",
        "2",
      ]
    }
    toggleTabs={null}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <NavItem
          onClick={null}
          tag="li"
          value="1"
        >
          <NavLink
            className="active"
            tag="a"
          >
            Season 1
          </NavLink>
        </NavItem>,
        <NavItem
          onClick={null}
          tag="li"
          value="2"
        >
          <NavLink
            className=""
            tag="a"
          >
            Season 2
          </NavLink>
        </NavItem>,
      ],
      "tabs": true,
      "tag": "ul",
      "vertical": false,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "children": <NavLink
            className="active"
            tag="a"
          >
            Season 1
          </NavLink>,
          "onClick": null,
          "tag": "li",
          "value": "1",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Season 1",
            "className": "active",
            "tag": "a",
          },
          "ref": null,
          "rendered": "Season 1",
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "function",
        "props": Object {
          "children": <NavLink
            className=""
            tag="a"
          >
            Season 2
          </NavLink>,
          "onClick": null,
          "tag": "li",
          "value": "2",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Season 2",
            "className": "",
            "tag": "a",
          },
          "ref": null,
          "rendered": "Season 2",
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <NavItem
            onClick={null}
            tag="li"
            value="1"
          >
            <NavLink
              className="active"
              tag="a"
            >
              Season 1
            </NavLink>
          </NavItem>,
          <NavItem
            onClick={null}
            tag="li"
            value="2"
          >
            <NavLink
              className=""
              tag="a"
            >
              Season 2
            </NavLink>
          </NavItem>,
        ],
        "tabs": true,
        "tag": "ul",
        "vertical": false,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "children": <NavLink
              className="active"
              tag="a"
            >
              Season 1
            </NavLink>,
            "onClick": null,
            "tag": "li",
            "value": "1",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Season 1",
              "className": "active",
              "tag": "a",
            },
            "ref": null,
            "rendered": "Season 1",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "function",
          "props": Object {
            "children": <NavLink
              className=""
              tag="a"
            >
              Season 2
            </NavLink>,
            "onClick": null,
            "tag": "li",
            "value": "2",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Season 2",
              "className": "",
              "tag": "a",
            },
            "ref": null,
            "rendered": "Season 2",
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
